src/mono/mono/mini/interp/transform.c

```diff
@@ -3549,10 +3549,20 @@ interp_transform_call (TransformData *td, MonoMethod *method, MonoMethod *target
 	}
 
 	CHECK_STACK_RET (td, csignature->param_count + csignature->hasthis, FALSE);
+
+	gboolean skip_tailcall = FALSE;
+	if (tailcall && target_method != NULL) {
```

Comment:
The virtual case is not really correctly handled by this. If we change the condition to `if (tailcall && !is_virtual && target_method != NULL) {`, would it still fix the crashes, for now ?

Reply:
Fixed :)